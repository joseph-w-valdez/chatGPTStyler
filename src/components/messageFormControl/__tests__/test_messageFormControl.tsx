import * as React from "react";
import { mount } from "enzyme";
import { MessageFormControl } from "../index"; // Adjust the import path based on your project structure

describe("MessageFormControl component", () => {
    it("renders correctly", () => {
        const wrapper = mount(
            <MessageFormControl section="Test" colorLiveChange={jest.fn()} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("updates colorType and colorCode correctly for HEX", () => {
        const wrapper = mount(
            <MessageFormControl section="Test" colorLiveChange={jest.fn()} />,
        );

        wrapper.find("select").simulate("change", { target: { value: "hex" } });
        wrapper
            .find("input")
            .simulate("change", { target: { value: "1a2b3c" } });

        expect(wrapper).toMatchSnapshot();
    });

    it("updates colorType and colorCode correctly for RGB", () => {
        const wrapper = mount(
            <MessageFormControl section="Test" colorLiveChange={jest.fn()} />,
        );

        wrapper.find("select").simulate("change", { target: { value: "rgb" } });
        wrapper
            .find("input")
            .simulate("change", { target: { value: "255, 0, 0" } });

        expect(wrapper).toMatchSnapshot();
    });
});
