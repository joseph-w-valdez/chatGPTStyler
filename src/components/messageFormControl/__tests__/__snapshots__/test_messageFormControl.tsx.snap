// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageFormControl renders correctly 1`] = `
<div>
  <h1>
    Test Message
  </h1>
  <hr />
  <label>
    Message Color
    <select
      defaultValue=""
      id="TestColorType"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Color Type
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="hex"
      >
        HEX
      </option>
      <option
        value="rgb"
      >
        RGB
      </option>
      <option
        value="hsl"
      >
        HSL
      </option>
    </select>
    <input
      disabled={true}
      id="TestTextColor"
      maxLength={0}
      placeholder="Select Color Type"
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`MessageFormControl updates colorType and colorCode correctly 1`] = `
<div>
  <h1>
    Test Message
  </h1>
  <hr />
  <label>
    Message Color
    <select
      defaultValue=""
      id="TestColorType"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Color Type
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="hex"
      >
        HEX
      </option>
      <option
        value="rgb"
      >
        RGB
      </option>
      <option
        value="hsl"
      >
        HSL
      </option>
    </select>
    <input
      disabled={false}
      id="TestTextColor"
      maxLength={6}
      placeholder="HEX"
      type="text"
      value="1a2b3c"
    />
  </label>
</div>
`;
